Refresh of hell due to incoming silly date format

#### Software and hardware versions 
Aggregate v2.0, but only for forms uploaded with Aggregate v1.x

#### Problem description
Aggregate will enter the refresh hell when processing dates in formats that include time offsets like `2018-12-14+01:00`

#### Steps to reproduce the problem
Yet to be determined

#### Expected behavior
Refresh hell shouldn't happen

#### Other information 
```
Caused by: java.time.format.DateTimeParseException: Text '1980-01-02+01:00' could not be parsed, unparsed text found at index 10
	at java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:1952) ~[na:1.8.0_181]
	at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1851) ~[na:1.8.0_181]
	at java.time.LocalDate.parse(LocalDate.java:400) ~[na:1.8.0_181]
	at java.time.LocalDate.parse(LocalDate.java:385) ~[na:1.8.0_181]
	at org.opendatakit.aggregate.submission.type.jr.JRTemporal.date(JRTemporal.java:63) ~[classes/:na]
	at org.opendatakit.aggregate.OptionalProduct$OptionalProduct2$Some.map(OptionalProduct.java:58) ~[classes/:na]
	at org.opendatakit.aggregate.submission.type.jr.JRDateType.getValueFromEntity(JRDateType.java:84) ~[classes/:na]
	at org.opendatakit.aggregate.submission.SubmissionSet.buildSubmissionFields(SubmissionSet.java:334) ~[classes/:na]
	at org.opendatakit.aggregate.submission.SubmissionSet.<init>(SubmissionSet.java:165) ~[classes/:na]
	at org.opendatakit.aggregate.submission.Submission.<init>(Submission.java:64) ~[classes/:na]
	at org.opendatakit.aggregate.query.submission.QueryByUIFilterGroup.populateSubmissions(QueryByUIFilterGroup.java:336) ~[classes/:na]
	at org.opendatakit.aggregate.server.SubmissionServiceImpl.getSubmissions(SubmissionServiceImpl.java:85) ~[classes/:na]
	at sun.reflect.GeneratedMethodAccessor18.invoke(Unknown Source) ~[na:na]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_181]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_181]
	at com.google.gwt.user.server.rpc.RPC.invokeAndEncodeResponse(RPC.java:587) ~[gwt-servlet-2.8.2.jar:na]
	... 61 common frames omitted
```

--- a/src/main/java/org/opendatakit/aggregate/submission/type/jr/JRDateType.java
+++ b/src/main/java/org/opendatakit/aggregate/submission/type/jr/JRDateType.java
@@ -69,7 +69,7 @@ public class JRDateType extends SubmissionSingleValueBase<JRTemporal> {
       raw = parsed
           .map(Date::toInstant)
           .map(i -> OffsetDateTime.ofInstant(i, ZoneId.systemDefault()))
-          .map(odt -> odt.format(DateTimeFormatter.ISO_DATE));
+          .map(odt -> odt.format(DateTimeFormatter.ISO_LOCAL_DATE));
     } else
       for (FormDataModel m : element.getFormDataModel().getChildren()) {
         switch (m.getOrdinalNumber().intValue()) {

--- a/src/main/java/org/opendatakit/aggregate/submission/type/jr/JRTemporal.java
+++ b/src/main/java/org/opendatakit/aggregate/submission/type/jr/JRTemporal.java
@@ -17,7 +17,7 @@
 package org.opendatakit.aggregate.submission.type.jr;
 
 import static java.time.ZoneId.systemDefault;
-import static java.time.format.DateTimeFormatter.ISO_DATE;
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE;
 import static java.time.format.DateTimeFormatter.ISO_OFFSET_DATE_TIME;
 import static java.time.format.DateTimeFormatter.ISO_OFFSET_TIME;
 import static java.util.Objects.requireNonNull;
@@ -56,7 +56,7 @@ public interface JRTemporal<T extends Temporal> {
 
   static JRTemporal date(Date parsed) {
     OffsetDateTime value = OffsetDateTime.ofInstant(requireNonNull(parsed).toInstant(), systemDefault()).truncatedTo(ChronoUnit.DAYS);
-    return new JRDate(Date.from(value.toInstant()), value.toLocalDate(), value.format(ISO_DATE));
+    return new JRDate(Date.from(value.toInstant()), value.toLocalDate(), value.format(ISO_LOCAL_DATE));
   }
 
   static JRTemporal date(Date parsed, String raw) {

